optimum/exporters/onnx/base.py

```diff
@@ -507,6 +517,8 @@ def __init__(
                 f"use_past = {use_past} is different than use_present_in_outputs = {use_present_in_outputs}, the value "
                 "of use_present_in_outputs value will be used for the outputs."
             )
+        self.is_merged = False
+        self.use_cache_branch = None
```

Comment:
What's the difference between `use_cache_branch` and `use_past` and `use_past_in_inputs` ? I mean that `use_cache_branch` must for the case of merged decoder, but why do we need to distinguish them?

And does `use_cache_branch` urges `use_past=True`?

Reply:
> does use_cache_branch urges use_past=True?

Yes, in other cases `use_cache_branch` does not make sense.

About the difference on `use_past` and `use_past_in_inputs`, it seems like code legacy that could be simplified. Or I miss something @michaelbenayoun ?

Reply:
`use_past` is the legacy here.
Basically you have two "use past": 

1. `use_past_in_inputs`: inputs will have past key values
2. `use_present_in_outputs`: outputs will have past key values

If you set only `use_past`, it sets both.